core/linker/default.h

```diff
@@ -55,21 +90,14 @@ ENTRY(main_entry)
 #endif /* defined (__ARM_FEATURE_CMSE) && (__ARM_FEATURE_CMSE == 3U) */
 #endif /* !defined(STACK_SIZE_NP) */
 
-/* Default uVisor own stack guard band
- * Note: Currently we do not actively use the stack guard to isolate the uVisor
- *       stack from the rest of the protected memories. For this reason the
- *       symbol is arbitrarily small and cannot be defined by the
- *       platform-specific configurations. */
-#define STACK_GUARD_BAND 32
-
 MEMORY
 {
   FLASH_NS (r x) : ORIGIN = (FLASH_ORIGIN + FLASH_OFFSET),
                    LENGTH = UVISOR_FLASH_LENGTH_MAX
   FLASH_S  (r x) : ORIGIN = (FLASH_ORIGIN + SECURE_ALIAS_OFFSET + FLASH_OFFSET),
                    LENGTH = UVISOR_FLASH_LENGTH_MAX
   RAM_S    (rwx) : ORIGIN = (SRAM_ORIGIN + SECURE_ALIAS_OFFSET + SRAM_OFFSET),
-                   LENGTH = UVISOR_SRAM_LENGTH_USED - (STACK_SIZE + STACK_GUARD_BAND) - (STACK_SIZE_NP ? (STACK_SIZE_NP + STACK_GUARD_BAND) : 0)
+                   LENGTH = UVISOR_SRAM_LENGTH_USED - TOTAL_STACK_SIZE - (STACK_SIZE_NP ? (STACK_SIZE_NP + STACK_GUARD_BAND) : 0)
   STACK_S  (rw ) : ORIGIN = ORIGIN(RAM_S) + LENGTH(RAM_S),
                    LENGTH = UVISOR_SRAM_LENGTH_PROTECTED - LENGTH(RAM_S)
 }
```

Comment:
@Patater @alzix 
On this code STACK_S length equals to TOTAL_STACK_SIZE.
When filling the STACK_S with data (some lines below, in .stack section) we add alignments on top of TOTAL_STACK_SIZE and can actually exceed the STACK_S length we declared before.
We can subtract the alignment size from RAM_S length but then we end up with taking memory from RAM_S in favor of STACK_S.
Can you advise?

Reply:
It makes sense to me to take memory from RAM_S to use for STACK_S.

Reply:
It means taking 0x1000 bytes from RAM which is a half of it.

On the other hand, we can leave the code as is since the linker protects us from overflowing memory sections. Thus, in case a problem occurs, we get notified by the linker.
Actually, now the align command does not add a single byte.

Reply:
I am not sure this is possible since it has to be aligned with the application linker script.
If we switch between STACK_S and RAM_S then the application linker script will have to be changed.
@Patater @alzix Please check me up on this one

Reply:
uVisor is just reserving a large chunk of RAM (that it may or may not use) from the application's perspective. So long as uVisor stays within that chunk, the application won't be affected.

Reply:
Is it indeed the case?

Inside STACK_S in **uvisor linker script** we reserve space but also have symbols, e.g. __uvisor_stack_start__.
Inside RAM_S in **uvisor linker script** we also import symbols, like .bss.

Inside **.uvisor.bss** section in **application linker script** we first have:
*__uvisor_bss_main_start = .;*
*KEEP(*(.keep.uvisor.bss.main))*
*__uvisor_bss_main_end = .;*
Then we have:
*__uvisor_bss_boxes_start = .;*
*KEEP(*(.keep.uvisor.bss.boxes))*
*__uvisor_bss_boxes_end = .;*
All these are going into RAM_S in the application.

So AFAIU, they must be aligned.
Am I missing so something?

Reply:
The application linker script is making sure the space reserved is kept and properly communicated to uVisor.

The application has no knowledge of where uVisor keeps its stack, but it knows the uVisor stack must be within `__uvisor_bss_main_start` and `__uvisor_bss_main_end`. So long as uVisor places its stack within `__uvisor_bss_main_start` and `__uvisor_bss_main_end`, the application won't be affected.

Reply:
I don't understand what you suggest.
Do you suggest simplifying the calculation of STACK_S and RAM_S length?
If so, I would prefer doing this in a separate PR as it is not related to the stack protection we are adding.

Reply:
At least a separate commit would be good. If it helps make the stack protection easier to add, it would fit just fine in this PR.

I'm fine with what you have as is now, and was just making an observation as to how the linker script could be improved.